### DISCUSSION TITLE
Feature request: toggle to exclude file extension from file name

### DISCUSSION THREAD

**Discussion by stevenlis** [PDConSec/vsc-print#311] [Ideas]
Hi @PeterWone

Could you please consider adding a toggle to exclude the file extension from the file name when printing various scripts and saving them as PDFs? For instance, a file named "file-name.py" currently results in a file name of "file-name.py.pdf." It would be more readable to simply have "file-name.pdf." Your thoughts on this would be appreciated. Thank you in advance!

**Comment by PeterWone** [comment]
This is one of those problems that's easy for a person and quite hard to code with 100% reliability, but we can certainly try. Again, after the build is fixed.

We can probably make it work for known cases and fall back to the current behaviour. Or your toggle suggestion would work too, I guess.

**Comment by stevenlis** [comment]
I know that VSCode can detect different languages. I wonder if you could leverage this feature in VSCode to remove the file extension from the file name when printing. For some unknown or lesser-known languages/file types, you can install an extension in VSCode to make them detectable sometimes. Just a thought.

![CS-Fi3vZBoA@2x](https://github.com/PDConSec/vsc-print/assets/33796896/c1cd4e14-7284-4472-a577-9fc211875315)
